src/charm.py

```diff
@@ -129,9 +128,22 @@ def _on_update_status(self, _):
     # END: hook functions
 
     # BEGIN: helper functions
+    def update_external_services(self) -> None:
+        """Attempts to update any external Kubernetes services."""
+        if not self.is_external_client:
+            return
+
+        # every unit attempts to create a nodeport service
+        # if exists, will silently continue
+        self.node_port_manager.apply_service(
+            service=self.node_port_manager.build_node_port_services(
+                port=Config.MONGOS_PORT
+            )
+        )
+
     def is_integrated_to_config_server(self) -> bool:
         """Returns True if the mongos application is integrated to a config-server."""
-        return self.model.relations[Config.Relations.CLUSTER_RELATIONS_NAME] is not None
+        return len(self.model.relations[Config.Relations.CLUSTER_RELATIONS_NAME])
```

Comment:
1. Does `self.model.relations[Config.Relations.CLUSTER_RELATIONS_NAME]` throw a KeyError or similar if there is not relation? or does it return an empty list? 
2. This breaks the contract of the method signature and may be error prone on the client code. I prefer to be explicit here (assuming it returns an empty list if no relation).  
```suggestion
        return len(self.model.relations[Config.Relations.CLUSTER_RELATIONS_NAME]) > 0
```

Reply:
> 1. Does self.model.relations[Config.Relations.CLUSTER_RELATIONS_NAME] throw a KeyError or similar if there is not relation? or does it return an empty list?

empty list

> This breaks the contract of the method signature and may be error prone on the client code. I prefer to be explicit here (assuming it returns an empty list if no relation).

Its actually changed in #20 - let me update it to be inline with that one